docs/en/ingest-management/fleet/air-gapped.asciidoc

```diff
@@ -0,0 +1,126 @@
+
+The {package-registry} can be deployed and hosted on-site using one of the
+available Docker images. These docker images include the {package-registry} and
+a selection of packages.
+
+There are different distributions available:
+
+* {version} (recommended): `docker.elastic.co/package-registry/distribution:{version}` - Selection of packages from the production repository released with the {version} version of the {stack}.
+* production: `docker.elastic.co/package-registry/distribution:production` - Packages available in the production registry (https://epr.elastic.co).
+* staging: `docker.elastic.co/package-registry/distribution:staging` - Packages available in the staging registry (https://epr-staging.elastic.co), they can be pending of validation.
+* snapshot: `docker.elastic.co/package-registry/distribution:snapshot` - Packages under development.
+
+ifeval::["{release-state}"=="unreleased"]
+[WARNING]
+====
+Version {version} of the {package-registry} distribution has not yet been released.
+====
+endif::[]
+
+If you want to update the distribution image, you need to re-pull the image and restart docker container.
+
+Every distribution contains packages that can be used by different versions of
+the {stack}. The {package-registry} API exposes a {kib} version constraint that
+allows for filtering packages that are compatible with a particular version.
+
+NOTE: These steps use the standard Docker CLI, but it shouldn't be hard to transform them into Kubernetes descriptor file.
+Here is the k8s descriptor used by the e2e-testing project: https://github.com/elastic/e2e-testing/blob/k8s-deployment/cli/config/kubernetes/base/package-registry/[yaml files].
+These images can also be used with other container runtimes compatible with Docker images.
+
+1. Pull the Docker image from the public Docker registry:
++
+[source,bash]
+----
+docker pull docker.elastic.co/package-registry/distribution:{version}
+----
+
+2. Save the Docker image locally:
++
+[source,bash]
+----
+docker save -o package-registry-{version}.tar docker.elastic.co/package-registry/distribution:{version}
+----
++
+TIP: please mind the image size, so you won't hit any capacity limit.
+
+3. Transfer the image to the air-gapped environment and load:
++
+[source,bash]
+----
+docker load -i package-registry-{version}.tar
+----
+
+4. Run the {package-registry}:
++
+[source,bash]
+----
+docker run -it docker.elastic.co/package-registry/distribution:{version}
+----
+
+5. (Optional) Define the internal healthcheck for the service as:
++
+[source,bash]
+----
+curl -f http://127.0.0.1:8080
+----
+
+[discrete]
+[[air-gapped-diy-epr-kibana]]
+=== Connect Kibana to the hosted Package Registry
+
+There is a dedicated property in the {kib} config to change the URL of the Package Registry's endpoint to a custom one.
+The example below connects to an internally hosted instance:
+
+[source,yaml]
+----
+xpack.fleet.registryUrl: "http://package-registry.corp.net:8080"
```

Comment:
Would it be possible to add in a note that this Kibana setting is only available in Enterprise license? Currently, this is not mentioned anywhere, but Kibana prints this out in the log if we configure this setting without an Enterprise license.